hail/src/main/scala/is/hail/expr/ir/Parser.scala

```diff
@@ -157,15 +157,7 @@ object IRParser {
   def error(t: Token, msg: String): Nothing = ParserUtils.error(t.pos, msg)
 
   def deserialize[T](str: String)(implicit formats: Formats, mf: Manifest[T]): T = {
-    try {
```

Comment:
this produces worse user errors...

Reply:
the "error summary" will be a mapping exception, I think, not a "your VCF has this problem"

Reply:
Before I removed this I got a mapping exception that said it found a "JNothing" when it was looking for "object". When I removed this, I got an exception that wrapped the mapping exception and mentioned a field name that was missing in the JSON (I had misspelled the field name in python).

Wrapping lower level exceptions with exceptions containing more context is very common in Java libraries. I know we do this kind of transformation frequently, but I'm *strongly* against it. This isn't the first time I've had to remove one our try-catches so as to find a meaningful error message.

Reply:
This won't change the error summary either way, we always take the deepest message (which in this case is next to useless). This just adds ~50 lines of stack trace.

Reply:
ah, OK, sure. Must be confusing this wrapping with something else!